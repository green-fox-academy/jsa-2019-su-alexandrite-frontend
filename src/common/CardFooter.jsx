import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

const CardFooter = ({ children }) => (
  <View style={styles.cardFooter}>
    {children}
  </View>
);

CardFooter.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
};

CardFooter.defaultProps = {
  children: [],
};

export default CardFooter;
