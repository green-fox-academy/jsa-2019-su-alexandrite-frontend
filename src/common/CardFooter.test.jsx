import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import CardFooter from './CardFooter';

test('CardFooter render properly', () => {
  const tree = renderer.create(
    <CardFooter />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('CardFooter render properly with children', () => {
  const tree = renderer.create(
    <CardFooter><Text>test children</Text></CardFooter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
