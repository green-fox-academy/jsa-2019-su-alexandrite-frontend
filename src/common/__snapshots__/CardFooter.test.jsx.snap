// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooter render properly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "paddingTop": 15,
    }
  }
/>
`;

exports[`CardFooter render properly with children 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "paddingTop": 15,
    }
  }
>
  <Text>
    test children
  </Text>
</View>
`;
